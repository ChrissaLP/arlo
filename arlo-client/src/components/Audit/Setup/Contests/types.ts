export interface IChoiceValues {
  id?: string
  name: string
  numVotes: string | number
}

export interface IContestValues {
  name: string
  isTargeted: boolean
  totalBallotsCast: string
  numWinners: string
  votesAllowed: string
  choices: IChoiceValues[]
}

export interface IValues {
  contests: IContestValues[]
}
