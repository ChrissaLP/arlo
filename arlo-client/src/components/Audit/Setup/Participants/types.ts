export interface IValues {
  state: string
}
