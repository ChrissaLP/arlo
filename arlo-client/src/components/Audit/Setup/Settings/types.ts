import { IAuditSettings } from '../../../../types'

export interface IValues {
  electionName: IAuditSettings['electionName']
  online: IAuditSettings['online']
  randomSeed: IAuditSettings['randomSeed']
  riskLimit: IAuditSettings['riskLimit']
}
