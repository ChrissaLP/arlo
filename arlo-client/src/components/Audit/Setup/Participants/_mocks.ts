const jurisdictionFile = new File(
  [
    'Jurisdiction","Admin Email"',
    '"Death Star","wtarkin@empire.gov"',
    '"Hoth","admin@rebelalliance.ninja"',
    '"Tatooine","jabba@hutt.biz"',
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
