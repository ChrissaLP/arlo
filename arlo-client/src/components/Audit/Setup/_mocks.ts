import { ElementType } from '../../../types'
import { setupStages } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
) => {
  const prevTitle = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    title: prevTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextTitle = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    title: nextTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  return { prevStage, nextStage }
}

export default relativeStages
