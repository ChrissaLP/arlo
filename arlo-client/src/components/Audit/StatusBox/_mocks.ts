import {
  IJurisdictionsResponse,
  FileProcessingStatus,
} from './getJurisdictions'
import { IRound } from '../../../types'

export const jurisdictionMocks: {
  [key in
    | 'none'
    | 'noUploads'
    | 'halfUploads'
    | 'allFinishedFirstRound']: IJurisdictionsResponse
} = {
  none: { jurisdictions: [] },
  noUploads: {
    jurisdictions: [
      {
        id: 'jurisdiction-1',
        name: 'Jurisdiction One',
        currentRoundStatus: null,
        ballotManifest: {
          file: null,
          numBallots: null,
          numBatches: null,
          processing: null,
        },
      },
      {
        id: 'jurisdiction-2',
        name: 'Jurisdiction Two',
        currentRoundStatus: null,
        ballotManifest: {
          file: null,
          numBallots: null,
          numBatches: null,
          processing: null,
        },
      },
    ],
  },
  halfUploads: {
    jurisdictions: [
      {
        id: 'jurisdiction-1',
        name: 'Jurisdiction One',
        currentRoundStatus: null,
        ballotManifest: {
          file: 'ballot manifest',
          numBallots: 10,
          numBatches: 1,
          processing: FileProcessingStatus.Processed,
        },
      },
      {
        id: 'jurisdiction-2',
        name: 'Jurisdiction Two',
        currentRoundStatus: null,
        ballotManifest: {
          file: null,
          numBallots: null,
          numBatches: null,
          processing: null,
        },
      },
    ],
  },
  allFinishedFirstRound: {
    jurisdictions: [
      {
        id: 'jurisdiction-1',
        name: 'Jurisdiction One',
        currentRoundStatus: 1,
        ballotManifest: {
          file: 'ballot manifest',
          numBallots: 10,
          numBatches: 1,
          processing: FileProcessingStatus.Processed,
        },
      },
      {
        id: 'jurisdiction-2',
        name: 'Jurisdiction Two',
        currentRoundStatus: 1,
        ballotManifest: {
          file: 'ballot manifest',
          numBallots: 10,
          numBatches: 1,
          processing: FileProcessingStatus.Processed,
        },
      },
    ],
  },
}

export const roundMocks: {
  [key in
    | 'none'
    | 'firstRoundComplete'
    | 'firstRoundIncomplete'
    | 'secondRoundIncomplete']: { rounds: IRound[] }
} = {
  none: { rounds: [] },
  firstRoundIncomplete: {
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: null,
              pvalue: null,
            },
            id: 'contest-1',
            results: {},
            sampleSize: null,
            sampleSizeOptions: null,
          },
        ],
        endedAt: null,
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-1',
      },
    ],
  },
  firstRoundComplete: {
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: true,
              pvalue: 0.00020431431380638307,
            },
            id: 'contest-1',
            results: {
              'choice-1': 100,
              'choice-2': 167,
            },
            sampleSize: 379,
            sampleSizeOptions: [
              { size: 269, type: 'ASN', prob: null },
              { size: 379, prob: 0.8, type: null },
              { size: 78, prob: null, type: null },
            ],
          },
        ],
        endedAt: '2019-07-18T16:59:34.000Z',
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-1',
      },
    ],
  },
  secondRoundIncomplete: {
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: false,
              pvalue: 1,
            },
            id: 'contest-1',
            results: {
              'choice-1': 0,
              'choice-2': 0,
            },
            sampleSize: null,
            sampleSizeOptions: [
              { size: 269, type: 'ASN', prob: null },
              { size: 379, prob: 0.8, type: null },
              { size: 78, prob: null, type: null },
            ],
          },
        ],
        endedAt: '2019-07-18T16:35:07.000Z',
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-1',
      },
      {
        contests: [
          {
            endMeasurements: {
              isComplete: null,
              pvalue: null,
            },
            id: 'contest-1',
            results: {},
            sampleSize: null,
            sampleSizeOptions: [
              { size: 269, type: 'ASN', prob: null },
              { size: 379, prob: 0.8, type: null },
              { size: 78, prob: null, type: null },
            ],
          },
        ],
        endedAt: null,
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-2',
      },
    ],
  },
}
