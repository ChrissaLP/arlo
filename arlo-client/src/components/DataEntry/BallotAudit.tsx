import React, { useState, useEffect } from 'react'
import { Formik, FormikProps, getIn, Field } from 'formik'
import { H4, H3, Button } from '@blueprintjs/core'
import {
  BallotRow,
  FormBlock,
  RadioGroupFlex,
  ProgressActions,
  FlushDivider,
} from './Atoms'
import BlockRadio from './BlockRadio'
import FormButton from '../Form/FormButton'
import { IReview, IContest } from '../../types'
import FormField from '../Form/FormField'

interface IProps {
  contest: IContest
  goReview: () => void
  review: IReview
  setReview: (arg0: { vote: string; comment: string }) => void
  previousBallot: () => void
}

interface IOptions {
  vote: string
  comment: string
}

const BallotAudit: React.FC<IProps> = ({
  contest,
  review,
  goReview,
  setReview,
  previousBallot,
}: IProps) => {
  const [commenting, setCommenting] = useState(false)
  useEffect(() => {
    setCommenting(!!review.comment)
  }, [review.comment])
  return (
    <BallotRow>
      <div className="ballot-side"></div>
      <div className="ballot-main">
        <H4>Instructions</H4>
        <p>
          Select <strong>all</strong> the candidates/choices below that you see
          marked on the paper ballot.
        </p>
        <p>
          If the voter did not vote in the contest, select &quot;Blank vote/no
          mark.&quot;
        </p>
        <p>
          If the audit board cannot agree, select &quot;No audit board
          consensus.&quot; You may add a comment for additional information
          about the disagreement.
        </p>
        <Formik
          initialValues={review}
          enableReinitialize
          onSubmit={values => {
            setReview(values)
            goReview()
          }}
          render={({
            handleSubmit,
            values,
            setFieldValue,
          }: FormikProps<IOptions>) => {
            const radioProps = {
              name: 'vote',
              handleChange: (e: React.ChangeEvent<HTMLInputElement>) =>
                setFieldValue('vote', e.currentTarget.value),
            }
            const toggleCommenting = () => {
              setCommenting(!commenting)
              setFieldValue('comment', '')
            }
            return (
              <>
                <FormBlock>
                  <H3>{contest.name}</H3>
                  <FlushDivider />
                  <RadioGroupFlex
                    name="vote"
                    onChange={
                      /* istanbul ignore next */
                      () => undefined
                    } // required by blueprintjs but we're implementing on BlockRadio instead
                    selectedValue={getIn(values, 'vote')}
                  >
                    {contest.choices.map(c => (
                      <BlockRadio
                        key={c.id}
                        {...radioProps}
                        checked={getIn(values, 'vote') === c.name}
                        value={c.name}
                      />
                    ))}
                    <BlockRadio
                      {...radioProps}
                      gray
                      checked={
                        getIn(values, 'vote') === "Audit board can't agree"
                      }
                      value="Audit board can't agree"
                    />
                    <BlockRadio
                      {...radioProps}
                      gray
                      checked={getIn(values, 'vote') === 'Blank vote/no mark'}
                      value="Blank vote/no mark"
                    />
                  </RadioGroupFlex>
                  <Button minimal icon="edit" onClick={toggleCommenting}>
                    {commenting ? 'Remove comment' : 'Add comment'}
                  </Button>
                  {commenting && (
                    <Field
                      name="comment"
                      type="textarea"
                      data-testid="comment-textarea"
                      component={FormField}
                    />
                  )}
                </FormBlock>
                <ProgressActions>
                  <FormButton
                    type="submit"
                    onClick={handleSubmit}
                    disabled={!values.vote}
                    intent="success"
                    data-testid={
                      !values.vote ? 'disabled-review' : 'enabled-review'
                    }
                  >
                    Review
                  </FormButton>
                  <Button onClick={previousBallot} minimal>
                    Back
                  </Button>
                </ProgressActions>
              </>
            )
          }}
        />
      </div>
    </BallotRow>
  )
}

export default BallotAudit
