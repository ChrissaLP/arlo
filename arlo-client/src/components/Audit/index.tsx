import React, { useState, useEffect, useCallback, useContext } from 'react'
import { useRouteMatch, RouteComponentProps } from 'react-router-dom'
import EstimateSampleSize from './EstimateSampleSize'
import SelectBallotsToAudit from './SelectBallotsToAudit'
import CalculateRiskMeasurement from './CalculateRiskMeasurement'
import { api, checkAndToast } from '../utilities'
import { IAudit, IErrorResponse, ElementType } from '../../types'
import ResetButton from './ResetButton'
import Wrapper from '../Atoms/Wrapper'
import Sidebar from '../Atoms/Sidebar'
import { AuthDataContext } from '../UserContext'
import Setup, { setupStages } from './Setup'
import useSetupMenuItems from './useSetupMenuItems'

const initialData: IAudit = {
  name: '',
  frozenAt: null,
  online: true,
  riskLimit: '',
  randomSeed: '',
  contests: [],
  jurisdictions: [],
  rounds: [],
  isMultiJurisdiction: false,
}

interface IParams {
  electionId: string
  view: 'setup' | 'progress'
}

const Audit: React.FC<{}> = () => {
  const match: RouteComponentProps<IParams>['match'] | null = useRouteMatch(
    '/election/:electionId/:view?'
  )
  /* istanbul ignore next */
  const viewMatch = match ? match.params.view : undefined
  /* istanbul ignore next */
  const electionId = match ? match.params.electionId : ''

  const { isAuthenticated, meta } = useContext(AuthDataContext)

  const [isLoading, setIsLoading] = useState<boolean>(false)

  const [audit, setAudit] = useState(initialData)

  const getStatus = useCallback(async (): Promise<IAudit> => {
    const auditStatusOrError: IAudit | IErrorResponse = await api(
      `/election/${electionId}/audit/status`
    )
    if (checkAndToast(auditStatusOrError)) {
      return initialData
    }
    return auditStatusOrError
  }, [electionId])

  const updateAudit = useCallback(async () => {
    const auditStatus = await getStatus()
    setIsLoading(true)
    setAudit(auditStatus)
    setIsLoading(false)
  }, [getStatus])

  useEffect(() => {
    updateAudit()
  }, [updateAudit])

  const showSelectBallotsToAudit =
    !audit.isMultiJurisdiction &&
    !!audit.contests.length &&
    audit.rounds[0].contests.every(c => !!c.sampleSizeOptions)
  const showCalculateRiskMeasurement =
    !!audit.rounds.length && audit.rounds[0].contests.every(c => !!c.sampleSize)

  const [stage, setStage] = useState<ElementType<typeof setupStages>>(
    'Participants'
  )

  const [menuItems, refresh] = useSetupMenuItems(stage, setStage, electionId)

  useEffect(() => {
    refresh()
  }, [refresh])

  return (
    <Wrapper className={!isAuthenticated ? 'single-page' : ''}>
      <ResetButton
        electionId={electionId}
        disabled={!audit.contests.length || isLoading}
        updateAudit={updateAudit}
      />

      {isAuthenticated &&
      (viewMatch === 'setup' || viewMatch === 'progress') ? (
        <>
          {meta!.type === 'audit_admin' && (
            <Sidebar title="Audit Setup" menuItems={menuItems} />
          )}
          <Setup stage={stage} menuItems={menuItems} />
        </>
      ) : (
        <>
          <EstimateSampleSize
            audit={audit}
            isLoading={isLoading && !showSelectBallotsToAudit}
            setIsLoading={setIsLoading}
            updateAudit={updateAudit}
            getStatus={getStatus}
            electionId={electionId}
          />

          {showSelectBallotsToAudit && (
            <SelectBallotsToAudit
              audit={audit}
              isLoading={isLoading && !showCalculateRiskMeasurement}
              setIsLoading={setIsLoading}
              updateAudit={updateAudit}
              getStatus={getStatus}
              electionId={electionId}
            />
          )}

          {showCalculateRiskMeasurement && (
            <CalculateRiskMeasurement
              audit={audit}
              isLoading={isLoading}
              setIsLoading={setIsLoading}
              updateAudit={updateAudit}
              getStatus={getStatus}
              electionId={electionId}
            />
          )}
        </>
      )}
    </Wrapper>
  )
}

export default Audit
