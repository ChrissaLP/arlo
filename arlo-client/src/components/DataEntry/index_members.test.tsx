import React from 'react'
import { wait } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { routerTestProps, asyncActRender } from '../testUtilities'
import DataEntry from './index'
import { dummyBoard, dummyBallots } from './_mocks'
import { statusStates } from '../Audit/_mocks'
import * as utilities from '../utilities'
import { IAudit, IBallot } from '../../types'

const memberlessDummy = statusStates[3]

const apiMock: jest.SpyInstance<
  ReturnType<typeof utilities.api>,
  Parameters<typeof utilities.api>
> = jest.spyOn(utilities, 'api').mockImplementation()

const memberingMock = async (
  endpoint: string
): Promise<IAudit | { ballots: IBallot[] }> => {
  switch (endpoint) {
    case '/election/1/audit/status':
      memberlessDummy.jurisdictions[0].auditBoards = [dummyBoard[0]]
      return memberlessDummy
    default:
      return dummyBallots
  }
}

afterEach(() => {
  apiMock.mockClear()
})

const routeProps = routerTestProps('/election/:electionId/board/:token', {
  electionId: '1',
  token: '123',
})

describe('DataEntry initial load', () => {
  beforeEach(() => {
    apiMock.mockImplementation(memberingMock)
  })

  it('renders correctly', async () => {
    const { container } = await asyncActRender(
      <Router {...routeProps}>
        <DataEntry {...routeProps} />
      </Router>
    )
    expect(container).toMatchSnapshot()
  })

  it('fetches initial state from api', async () => {
    const { container } = await asyncActRender(
      <Router {...routeProps}>
        <DataEntry {...routeProps} />
      </Router>
    )
    await wait(() => {
      expect(apiMock).toBeCalled()
      expect(container).toMatchSnapshot()
    })
  })

  it('renders member form', async () => {
    const { container, getByText } = await asyncActRender(
      <Router {...routeProps}>
        <DataEntry {...routeProps} />
      </Router>
    )
    await wait(() => {
      expect(apiMock).toBeCalled()
      expect(
        getByText('Member Sign in for Audit Board: Audit Board #1')
      ).toBeTruthy()
      expect(container).toMatchSnapshot()
    })
  })
})
