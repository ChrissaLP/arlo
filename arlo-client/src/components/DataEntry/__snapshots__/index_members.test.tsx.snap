// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataEntry initial load fetches initial state from api 1`] = `
<div>
  <div
    class="sc-kEYyzF dLLIBs"
  >
    <h1
      class="bp3-heading"
    >
      Member Sign in for Audit Board: 
      Audit Board #1
    </h1>
    <p>
      Enter the information below for each member of 
      Jurisdiction 1
       
      Audit Board #1
       below, then click "Next" to proceed.
    </p>
    <div
      class="sc-ifAKCX dUfgxM"
    >
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      />
      <form>
        <div
          class="sc-EHOje gNDuQv"
        >
          <h3
            class="bp3-heading sc-gzVnrw hYCxik"
          >
            Audit Board Member
          </h3>
          <label
            for="[0]name"
          >
            <span
              class="sc-iwsKbI haXKLx"
            >
              Full Name
            </span>
            <input
              class="sc-gZMcBi bZTBmE"
              id="[0]name"
              name="[0]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-iwsKbI haXKLx"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="OTH"
                />
                <span
                  class="bp3-control-indicator"
                />
                Other
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[0]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <div
          class="sc-EHOje gNDuQv"
        >
          <h3
            class="bp3-heading sc-gzVnrw hYCxik"
          >
            Audit Board Member
          </h3>
          <label
            for="[1]name"
          >
            <span
              class="sc-iwsKbI haXKLx"
            >
              Full Name
            </span>
            <input
              class="sc-gZMcBi bZTBmE"
              id="[1]name"
              name="[1]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-iwsKbI haXKLx"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="OTH"
                />
                <span
                  class="bp3-control-indicator"
                />
                Other
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[1]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-dnqmqq iQfLYL"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Next
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`DataEntry initial load renders correctly 1`] = `
<div>
  <div
    class="sc-kEYyzF dLLIBs"
  >
    <h1
      class="bp3-heading"
    >
      Member Sign in for Audit Board: 
      Audit Board #1
    </h1>
    <p>
      Enter the information below for each member of 
      Jurisdiction 1
       
      Audit Board #1
       below, then click "Next" to proceed.
    </p>
    <div
      class="sc-ifAKCX dUfgxM"
    >
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      />
      <form>
        <div
          class="sc-EHOje gNDuQv"
        >
          <h3
            class="bp3-heading sc-gzVnrw hYCxik"
          >
            Audit Board Member
          </h3>
          <label
            for="[0]name"
          >
            <span
              class="sc-iwsKbI haXKLx"
            >
              Full Name
            </span>
            <input
              class="sc-gZMcBi bZTBmE"
              id="[0]name"
              name="[0]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-iwsKbI haXKLx"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="OTH"
                />
                <span
                  class="bp3-control-indicator"
                />
                Other
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[0]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <div
          class="sc-EHOje gNDuQv"
        >
          <h3
            class="bp3-heading sc-gzVnrw hYCxik"
          >
            Audit Board Member
          </h3>
          <label
            for="[1]name"
          >
            <span
              class="sc-iwsKbI haXKLx"
            >
              Full Name
            </span>
            <input
              class="sc-gZMcBi bZTBmE"
              id="[1]name"
              name="[1]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-iwsKbI haXKLx"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="OTH"
                />
                <span
                  class="bp3-control-indicator"
                />
                Other
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[1]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-dnqmqq iQfLYL"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Next
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`DataEntry initial load renders member form 1`] = `
<div>
  <div
    class="sc-kEYyzF dLLIBs"
  >
    <h1
      class="bp3-heading"
    >
      Member Sign in for Audit Board: 
      Audit Board #1
    </h1>
    <p>
      Enter the information below for each member of 
      Jurisdiction 1
       
      Audit Board #1
       below, then click "Next" to proceed.
    </p>
    <div
      class="sc-ifAKCX dUfgxM"
    >
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      />
      <form>
        <div
          class="sc-EHOje gNDuQv"
        >
          <h3
            class="bp3-heading sc-gzVnrw hYCxik"
          >
            Audit Board Member
          </h3>
          <label
            for="[0]name"
          >
            <span
              class="sc-iwsKbI haXKLx"
            >
              Full Name
            </span>
            <input
              class="sc-gZMcBi bZTBmE"
              id="[0]name"
              name="[0]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-iwsKbI haXKLx"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[0]affiliation"
                  type="radio"
                  value="OTH"
                />
                <span
                  class="bp3-control-indicator"
                />
                Other
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[0]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <div
          class="sc-EHOje gNDuQv"
        >
          <h3
            class="bp3-heading sc-gzVnrw hYCxik"
          >
            Audit Board Member
          </h3>
          <label
            for="[1]name"
          >
            <span
              class="sc-iwsKbI haXKLx"
            >
              Full Name
            </span>
            <input
              class="sc-gZMcBi bZTBmE"
              id="[1]name"
              name="[1]name"
              value=""
            />
          </label>
          <div>
            <span
              class="sc-iwsKbI haXKLx"
            >
              Party Affiliation 
              <i>
                (optional)
              </i>
            </span>
            <div>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="DEM"
                />
                <span
                  class="bp3-control-indicator"
                />
                Democrat
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="REP"
                />
                <span
                  class="bp3-control-indicator"
                />
                Republican
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="LIB"
                />
                <span
                  class="bp3-control-indicator"
                />
                Libertarian
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="IND"
                />
                <span
                  class="bp3-control-indicator"
                />
                Independent/Unaffiliated
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  name="[1]affiliation"
                  type="radio"
                  value="OTH"
                />
                <span
                  class="bp3-control-indicator"
                />
                Other
              </label>
              <label
                class="bp3-control bp3-radio"
              >
                <input
                  checked=""
                  name="[1]affiliation"
                  type="radio"
                  value=""
                />
                <span
                  class="bp3-control-indicator"
                />
                None
              </label>
            </div>
          </div>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-dnqmqq iQfLYL"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Next
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
